big_tests/tests/connect_SUITE.erl

```diff
@@ -618,6 +632,38 @@ replaced_session_cannot_terminate(Config) ->
 
     escalus_connection:stop(Alice2).
 
+return_proper_stream_error_if_service_is_not_hidden(_Config) ->
+    % GIVEN MongooseIM is running default configuration
+    % WHEN we send non-XMPP payload
+    % THEN the server replies with stream error xml-not-well-formed and closes the connection
+    SendMalformedDataStep = fun(Client, Features) ->
+                                    escalus_connection:send_raw(Client, <<"malformed">>),
+                                    {Client, Features}
+                            end,
+    {ok, Connection, _} = escalus_connection:start([], [SendMalformedDataStep]),
+    escalus_connection:receive_stanza(Connection, #{ assert => is_stream_start }),
+    StreamErrorAssertion = {is_stream_error, [<<"xml-not-well-formed">>, <<>>]},
+    escalus_connection:receive_stanza(Connection, #{ assert => StreamErrorAssertion }),
+    %% Sometimes escalus needs a moment to report the connection as closed
+    mongoose_helper:wait_until(fun() -> escalus_connection:is_connected(Connection) end, false).
```

Comment:
There is `escalus_connection:wait_for_close/2` function which could be used here.